### PR TITLE
Remove unnecessary `activationEvents` from extension's package.json

### PR DESCRIPTION
**Pull request by norascheuch**
From VS Code version 1.74.0 onwards, we no longer need some activations events.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by koesie10 — COMMENTED** [review]
To do this, we first need to ensure we only support VS Code version 1.74.0 and up. I've placed a link in the internal issue to the process we need to follow to do that.

File: extensions/ql-vscode/package.json [hunk 1]
```diff
@@ -34,27 +34,8 @@
     }
   },
   "activationEvents": [
-    "onLanguage:ql",
     "onLanguage:ql-summary",
```
Comment (koesie10) [inline extensions/ql-vscode/package.json:37]: Can this also be removed? It seems like this is also included in our `contributes.languages`.

Comment (norascheuch) [reply extensions/ql-vscode/package.json:37]: Yes indeed! In https://code.visualstudio.com/api/references/activation-events#onLanguage it's stated at some point. Thank you! 
It's 'interesting' that we activate the extension when the evaluator-log.summary is opened (why?..) by defining a 'language' that is connected to this file.

Comment (koesie10) [reply extensions/ql-vscode/package.json]: It seems like we [do have some language support for summary files](https://github.com/github/vscode-codeql/blob/d0df2966c52e4f25124b59ca1a28717461c3ef14/extensions/ql-vscode/src/log-insights/summary-language-support.ts), so this is probably the easiest way to actually make VS Code use that language.

**Review by koesie10 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**